### Задание 1

Группа студентов изучает 10 различных дисциплин. Сколькими способами можно составить расписание на понедельник, если в этот день должно быть 4 разных занятия?

Порядок занятий неважен, значит нам нужно сочетание любых 4х занятий из 10, следовательно, надо выбрать сочетания

$$C_n^k = \frac{n!}{k!(n - k)!}$$

In [1]:
from math import factorial

In [2]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [3]:
combinations(10, 4)

210

Ответ: 210 способов

### Задание 2

Из колоды в 52 карты вынимают случайным образом 4 карты. Найти число исходов, соответствующих тому, что был вытянут хотя бы один туз.

Нас устраивает ситуация когда мы вытаскиваем 1, 2, 3 или 4 туза из колоды карт.

Сначала рассмотрим ситуацию выбора 1, 2, 3 или 4 тузов из 4х тузов

In [4]:
combinations(4,1)

4

In [5]:
combinations(4,2)

6

In [7]:
combinations(4,3)

4

In [8]:
combinations(4,4)

1

В первом случае нам надо выбрать оставшиеся 3 карты из 48

In [9]:
combinations(48,3)

17296

Во втором случае нам надо выбрать оставшиеся 2 карты из 48

In [10]:
combinations(48,2)

1128

В третьем случае - 1 карту из 48

In [11]:
combinations(48,1)

48

В последнем - 0 из 48

In [12]:
combinations(48,0)

1

Итог для каждого случая:

In [13]:
 4*17296

69184

In [14]:
6*1128

6768

In [21]:
4*48

192

In [16]:
1*1

1

Складываем полученные исходы:

In [17]:
69184+6768+192+1

76145

Ответ: 76145 исходов

### Задание 3

Семь человек рассаживаются наудачу на скамейке. Какова вероятность того, что два определённых человека будут сидеть рядом?

Порядок важен, поэтому сразу можем отметить, что нам понадобятся размещения

$$A_n^k = \frac{n!}{(n - k)!}$$

In [18]:
def arrangements(n, k):
    return int(factorial(n) / factorial(n - k))

Разместить два человека рядом в одном ряду можно только двумя способами, когда перый человек будет справа(1) или слева(2) от второго

In [20]:
arrangements(7,2)

42

А количество способов разместить 7 человек в ряд определяется перестановками

In [22]:
factorial(7)

5040

Итоговая вероятность равна количеству устраивающих нас исходов на общее количество исходов 

In [23]:
42/5040

0.008333333333333333

Ответ: Вероятность 8,3 %

### Задание 4

Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает: а) все? б) два?

Найдём число способов вытащить 3 любых билета из 60. Это бдет общее число исходов. Порядок не важен, поэтому используем сочетания

In [24]:
combinations(60,3)

34220

Но студент знает только 50 билетов, поэтому для благоприятного исхода нам надо вытащить любые 3 билета из 50

In [25]:
combinations(50,3)

19600

Тогда вероятность будет определяться отношением благоприятных исходов ко всем исходам

In [26]:
19600/34220

0.5727644652250146

Ответ (для 3х вопросов): вероятность 57,28%

Для второго случая число общих исходов будет таким же

Для благопрятного исхода студенту надо вытянуть 2 из 50 выученных

In [28]:
combinations(50,2)

1225

И любой 1 билет из оставшихся 10

In [29]:
combinations(10,1)

10

Общее количество благоприятных исходов

In [31]:
1225*10

12250

Итоговая вероятность

In [32]:
12250/34220

0.35797779076563413

Ответ (для 2х из 3х вопросов): 35,8%

### Задание 5

Бросается игральная кость. Пусть событие A - появление чётного числа, событие B - появление числа больше трёх. Являются ли эти события независимыми?

Если мыслить логически, то событие В с вероятностью 66% приведёт к событию А. Для события В, число больше 3х для игральной кости - это 4, 5 или 6. 5 и 6 - это чётные числа, значит в этом случае произошло событие А.  
Для собития А подходят числа 2, 4 и 6. С вероятностью 66%, если произошло событие А, то произошло и событие В, т.к. числа 4 и 6 подходят для этого события.  
Получается, что события зависимы (к тому же мы бросоем одну игральную кость).